backend/controllers/passController.js

```diff
@@ -0,0 +1,90 @@
+const asyncHandler = require("express-async-handler")
+require("dotenv").config()
+const { connectDB } = require("../config/db")
+const User = require("../model/user.model")
+const Pass = require("../model/pass.model")
+const { ErrorHandler } = require("../middleware/errorMiddleware")
+
+//@desc Generate a pass by user
+//@route POST /users/generatePass
+const generatePass = asyncHandler(async (req, res, next) => {
+  try {
+    await connectDB()
+    const { name, email, phone, duration, start } = req.body
+
+    const user = await User.findOne({ _id: req.user._id })
+    if (user) {
+      const newPass = await Pass.create({
+        name,
+        phone,
+        email,
+        duration,
+        start,
+        checkInStatus: false,
+        generatedUserId: user.id,
+        userName: req.user.name,
+      })
+      if (newPass) {
```

Comment:
Why we need this condition?

Reply:
for pass generation, so that msg for successfull pass generation is sent only if valid value is stored in newPass otherwise not

Reply:
Can you try to store invalid values and check what happens ?

1. Does the execution goes to line 27(above the if condition)? If not, where it goes and what happens after that?

Reply:
With invalid values, it shows status code 400.
The execution does not go to line 27 (checked by adding a console.log before line 27, it is not displayed)
The control goes to errorMiddleware from where it consoles the error to the terminal
`expose: true,
  statusCode: 400,
  status: 400,
  body: '{\n' +
    '    "name": "Urban Clap 240",\n' +
    '    "phone": "4545456767",\n' +
    '    "email" : "urbanclap220@gmail.com",\n' +
    '    "duration": 3ab,\n' +
    '    "start": "02-12-2022"\n' +
    '}',
  type: 'entity.parse.failed'
}`

Reply:
So, that means we can remove the condition and confidently say that if newPass is invalid in any case, then it will directly go to catch.